redfish-core/lib/pcie_slots.hpp

```diff
@@ -605,14 +558,50 @@ inline void handlePCIeSlotCollectionGet(
         return;
     }
 
-    constexpr std::array<std::string_view, 1> interfaces{
-        "xyz.openbmc_project.Inventory.Item.PCIeSlot"};
-    dbus::utility::getSubTree(
-        "/xyz/openbmc_project/inventory", 0, interfaces,
+    redfish::chassis_utils::getValidChassisPath(
+        asyncResp, chassisID,
         [asyncResp,
-         chassisID](const boost::system::error_code ec,
-                    const dbus::utility::MapperGetSubTreeResponse& subtree) {
-        onMapperSubtreeDone(asyncResp, chassisID, ec, subtree);
+         chassisID](const std::optional<std::string>& validChassisPath) {
+        if (!validChassisPath)
+        {
+            BMCWEB_LOG_ERROR << "Not a valid chassis ID:" << chassisID;
+            messages::resourceNotFound(asyncResp->res, "Chassis", chassisID);
+            return;
+        }
+        doHandlePCIeSlotCollectionGet(asyncResp, chassisID);
+        });
+}
+
+inline void doHandlePCIeSlotPatch(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp, size_t total,
+    const std::string& chassisID,
+    const std::map<unsigned int, bool>& locationIndicatorActiveMap)
+{
+    getValidPCIeSlotList(
+        asyncResp, chassisID,
+        [asyncResp, chassisID, total,
+         locationIndicatorActiveMap{std::move(locationIndicatorActiveMap)}](
+            const std::vector<std::pair<std::string, std::string>>&
+                slotPathConnNames) {
+        if (slotPathConnNames.size() != total)
+        {
+            BMCWEB_LOG_ERROR
+                << "The actual number of pcieslots is different from the number of the input slots";
+            messages::internalError(asyncResp->res);
```

Comment:
This is a PATCH.. I think we should return a 404 
For 1020 we returned a resourceNotFound 
https://github.com/ibm-openbmc/bmcweb/blob/1020/redfish-core/lib/pcie_slots.hpp#LL646C1-L648C60

Reply:
You could use 

        "ArraySizeTooLong": {
        
or 
        
                "ArraySizeTooShort": {
                
                
or 

        "InvalidIndex": {

Reply:
I think - InvalidIndex seems better because it represents the inconsistent count of array inputs to the valid/active pcie slots.